File: core/tmp/Python/array/maximum_points_you_can_obtain_from_cards.py
from typing import List

def maxScore(cardPoints: List[int], k: int) -> int:
    length = len(cardPoints)
    if length == k:
        return sum(cardPoints)

    front, back = [cardPoints[0]], [cardPoints[-1]]
    for i in range(1, length):
        front.append(front[i - 1] + cardPoints[i])
        back.append(back[i - 1] + cardPoints[length - 1 - i])
    
    result = max(front[k - 1], back[k - 1])
    flag = k - 2
    while flag >= 0:
        result = max(result, front[flag] + back[k - 2 - flag])
        flag -= 1
    
    return result

def maxScore_slid(cardPoints: List[int], k: int) -> int:
    length = len(cardPoints)
    s = sum(cardPoints[:length - k])
    min_sum = s
    for i in range(length - k, length):
        s += cardPoints[i] - cardPoints[i - length + k]
        min_sum = min(min_sum, s)
    return sum(cardPoints) - min_sum

if __name__ == "__main__":
    print(maxScore_slid(cardPoints = [1,79,80,1,1,1,200,1], k = 3))

File: core/tmp/Java/src/com/dbc/algorithm/Array/MaximumPointsYouCanObtainFromCards.java
package com.dbc.algorithm.Array;

public class MaximumPointsYouCanObtainFromCards {
    public static int maxScore(int[] cardPoints, int k) {
        int[] front = new int[cardPoints.length];
        int[] back = new int[cardPoints.length];

        front[0] = cardPoints[0];
        back[0] = cardPoints[cardPoints.length - 1];
        for (int i = 1; i < cardPoints.length; i++) {
            front[i] = front[i - 1] + cardPoints[i];
            back[i] = back[i - 1] + cardPoints[cardPoints.length - i - 1];
        }

        int result = Math.max(front[k - 1], back[k - 1]);
        int flag = k - 2;
        while (flag >= 0) {
            result = Math.max(result, front[flag] + back[k - 2 - flag]);
            flag--;
        }
        return result;
    }

    public static int maxScoreSlid(int[] cardPoints, int k) {
        int sum = 0, total = 0;
        for (int i = 0; i < cardPoints.length; i++) {
            if (i < cardPoints.length - k)
                sum += cardPoints[i];
            total += cardPoints[i];
        }

        int result = sum;
        for (int i = cardPoints.length - k; i < cardPoints.length; i++) {
            sum += cardPoints[i] - cardPoints[i - cardPoints.length + k];
            result = Math.min(result, sum);
        }

        return total - result;
    }
}
